doc/specs/stdlib_math.md

```diff
@@ -90,3 +90,188 @@ program demo_clip_real
   ! clipped_value <- 3.02500010
 end program demo_clip_real
 ```
+
+### `linspace` - Create a linearly spaced rank one array
+
+#### Description
+
+Returns a linearly spaced rank 1 array from [`start`, `end`]. Optionally, you can specify the length of the returned array by passing `n`.
+
+#### Syntax
+
+`res = [[stdlib_math(module):linspace(interface)]] (start, end [, n])`
+
+#### Status
+
+Experimental
+
+#### Class
+
+Function.
+
+#### Argument(s)
+
+`start`: Shall be scalar of any numeric type or kind. This argument is `intent(in)`.
+`end`: Shall be the same `type` and `kind` as `start`. This argument is `intent(in)`.
+`n`: Shall be an integer specifying the length of the output. This argument is `optional` and `intent(in)`.
+
+#### Output value or Result value
+
+The output is a rank 1 array whose length is either 100 (default value) or `n`.
+
+If `n` == 1, return a rank 1 array whose only element is `end`.
+If `n` <= 0, return a rank 1 array with length 0.
+
+If `start`/`end` are real or complex types, the `result` will be the same `type` and `kind` as `start`/`end`.
+If `start`/`end` are integer types, the `result` will default to a double precision real array.
```

Comment:
I'm confused. You say that `dp` might not refer to `double precision`, but then you are the one suggesting replacing `double precision` with `real(dp)`. Am I missing something here?

Reply:
In `stdlib_math`, the result is declared as `real(dp) :: res(n)`, with `dp` being provided by `stdlib_kinds`. `dp` is the same as `real64` provided by `iso_fortran_env` [here](https://github.com/fortran-lang/stdlib/blob/492543d0086ff3821ecf2ee8f0a9cf46a945a7ac/src/stdlib_kinds.f90#L3). However, `real64` does not garanty that it is `double precision`. See [this post by Doctor Fortran](https://stevelionel.com/drfortran/2017/03/27/doctor-fortran-in-it-takes-all-kinds/) for more details.

Therefore, I think it is more correct to use `real(dp)` in your sentence, instead of `double precision`.

Reply:
Awesome! Thanks for the clarification, I didn't understand the nuance between `dp` and `douple precision` but what you've said here clears that up. Currently incorporating your corrections :)

Reply:
Also, I'm sorry to keep bugging but can you approve the workflow checks? And if all of them pass, would you kindly merge this PR? I'm just so excited to be a first-time contributor!!! This is totally a tangent but as I mentioned I'll be starting a Comp Sci program in the fall, and this will be my first symbolic step in my programming career! And I can't wait!!!